server/src/test/java/org/apache/druid/discovery/BaseNodeRoleWatcherTest.java

```diff
@@ -125,15 +190,40 @@ private DiscoveryDruidNode buildDiscoveryDruidNode(NodeRole role, String host)
     );
   }
 
-  private void assertListener(TestListener listener, boolean nodeViewInitialized, List<DiscoveryDruidNode> nodesAdded, List<DiscoveryDruidNode> nodesRemoved)
+  private void assertListenerTimedOut(TestListener listener)
```

Comment:
why is it done inside exec?

Reply:
This was done to ensure the order of execution in tests.
Thanks for your suggestion of using the following implementation instead:
```
return new ScheduledThreadPoolExecutor(corePoolSize, Execs.makeThreadFactory(nameFormat))
        {
          @Override
          public Future<?> submit(Runnable task) {
            task.run();
            return Futures.immediateFuture(null);
          }
        };
```